### ISSUE TITLE
trial.test.test_directoryNotPackage is incorrect

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6400]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6400|
|Type|defect|
|Created|2013-04-02 18:56:13Z|
|Branch|https://github.com/twisted/twisted/tree/rm-directory-6400|

`trial.test.test_directoryNotPackage` was last changed in r14796 where an empty folder in `trial.test.directory` was also added. Deleting that `directory` doesn't fail the test so the test should be fixed (and also use a temporary directory).

<details><summary>Searchable metadata</summary>

```
trac-id__6400 6400
type__defect defect
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_rm_directory_6400 branches/rm-directory-6400
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1364928973000000 1364928973000000
changetime__1365784473000000 1365784473000000
version__None None
owner__thijs thijs
cc__thijs cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [37916]) Branching to 'rm-directory-6400'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [37917]) improve tests, add news file. refs [#6400](https://github.com/twisted/twisted/issues/6400)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

 * Added docstrings to tests
 * removed empty test 'directory'
 * fixed `test_directoryNotPackage` by creating a temp dir and checking for the correct error value
 * turned out `test_filenameNotPython` was referencing a non-existing file (`nonpython.py` which should `nonpython`) and after changing this it failed with a `SyntaxError` raised while importing that text file. I changed the test to expect that `SyntaxError`. It might be a bug but that could be handled in a different ticket.
 * added reference to `test_loader` in `trial.test.packages`

Build [here](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/rm-directory-6400).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

1.Is there a reason that `test_notFile` was changed to not use `assertRaises`? (which returns the exception, for doing further assertions)
 2. Consider using `FilePath.createDirectory` in `test_directoryNotPackage`
 3. Consider using `FilePath.setContent` in `test_filenameMatchesPackage`
 4. If I change `test_directoryNotPackage` to create a `__init__.py` in the directory, it still passes, so it isn't clear to me that the test is now correct.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [38038]) address review comments, refs [#6400](https://github.com/twisted/twisted/issues/6400)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [tom.prince](#note_5):
>  1. Is there a reason that `test_notFile` was changed to not use `assertRaises`? (which returns the exception, for doing further assertions)
Thanks, fixed.

>  2. Consider using `FilePath.createDirectory` in `test_directoryNotPackage`
>  3. Consider using `FilePath.setContent` in `test_filenameMatchesPackage`

Fixed, used FilePath wherever possible.

>  4. If I change `test_directoryNotPackage` to create a `__init__.py` in the directory, it still passes, so it isn't clear to me that the test is now correct.

Not sure what was wrong before, but adding a `emptyDir.child('__init__.py').setContent('')` now does fail the test as expected. Also moved creation of the dir to a temporary path (before it was actually created in source tree).

New build [here](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/rm-directory-6400), up for review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

Looks good. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [38048]) Merge rm-directory-6400: Improve test_loader.FileTest.

Author: thijs
Reviewer: tom.prince
Fixes: [#6400](https://github.com/twisted/twisted/issues/6400)
